src/system_database.ts

```diff
@@ -1189,79 +1163,82 @@ export class PostgresSystemDatabase implements SystemDatabase {
+        `${schemaName}.workflow_inputs.workflow_uuid`,
+      )
+      .orderBy(`${schemaName}.workflow_status.created_at`, input.sortDesc ? 'desc' : 'asc');
     if (input.workflowName) {
-      query = query.where('name', input.workflowName);
+      query = query.where(`${schemaName}.workflow_status.name`, input.workflowName);
+    }
+    if (input.workflow_id_prefix) {
+      query = query.whereLike(`${schemaName}.workflow_status.workflow_uuid`, `${input.workflow_id_prefix}%`);
     }
     if (input.workflowIDs) {
-      query = query.whereIn('workflow_uuid', input.workflowIDs);
+      query = query.whereIn(`${schemaName}.workflow_status.workflow_uuid`, input.workflowIDs);
     }
     if (input.authenticatedUser) {
-      query = query.where('authenticated_user', input.authenticatedUser);
+      query = query.where(`${schemaName}.workflow_status.authenticated_user`, input.authenticatedUser);
     }
     if (input.startTime) {
-      query = query.where('created_at', '>=', new Date(input.startTime).getTime());
+      query = query.where(`${schemaName}.workflow_status.created_at`, '>=', new Date(input.startTime).getTime());
     }
     if (input.endTime) {
-      query = query.where('created_at', '<=', new Date(input.endTime).getTime());
+      query = query.where(`${schemaName}.workflow_status.created_at`, '<=', new Date(input.endTime).getTime());
     }
     if (input.status) {
-      query = query.where('status', input.status);
+      query = query.where(`${schemaName}.workflow_status.status`, input.status);
     }
     if (input.applicationVersion) {
-      query = query.where('application_version', input.applicationVersion);
+      query = query.where(`${schemaName}.workflow_status.application_version`, input.applicationVersion);
     }
     if (input.limit) {
       query = query.limit(input.limit);
     }
     if (input.offset) {
       query = query.offset(input.offset);
     }
-    const rows = await query.select('workflow_uuid');
-    const workflowUUIDs = rows.map((row) => row.workflow_uuid);
-    return {
-      workflowUUIDs: workflowUUIDs,
-    };
+    const rows = await query;
+    return rows.map(PostgresSystemDatabase.#mapWorkflowStatus);
   }
 
-  async getQueuedWorkflows(input: GetQueuedWorkflowsInput): Promise<GetWorkflowsOutput> {
+  async getQueuedWorkflows(input: GetQueuedWorkflowsInput): Promise<WorkflowStatusInternal[]> {
+    const schemaName = DBOSExecutor.systemDBSchemaName;
+
     const sortDesc = input.sortDesc ?? false; // By default, sort in ascending order
-    let query = this.knexDB(`${DBOSExecutor.systemDBSchemaName}.workflow_queue`)
-      .join(
-        `${DBOSExecutor.systemDBSchemaName}.workflow_status`,
-        `${DBOSExecutor.systemDBSchemaName}.workflow_queue.workflow_uuid`,
-        '=',
-        `${DBOSExecutor.systemDBSchemaName}.workflow_status.workflow_uuid`,
+    let query = this.knexDB<workflow_status>(`${schemaName}.workflow_status`)
+      .join<workflow_inputs>(
```

Comment:
We should query queue and inner join on status --- the queue table is typically much smaller